go/service/identify.go

```diff
@@ -147,8 +149,234 @@ func (h *IdentifyHandler) resolveUserOrTeam(ctx context.Context, arg string) (u
 	return res.UserOrTeam(), nil
 }
 
+func (h *IdentifyHandler) resolveUser(ctx context.Context, assertion string) (u keybase1.User, err error) {
+
+	res := h.G().Resolver.ResolveFullExpressionNeedUsername(ctx, assertion)
+	err = res.GetError()
+	if err != nil {
+		return u, err
+	}
+	u = res.User()
+	if !u.Uid.Exists() {
+		return u, fmt.Errorf("no resolution for: %v", assertion)
+	}
+	return u, nil
+}
+
 func (h *IdentifyHandler) ResolveIdentifyImplicitTeam(ctx context.Context, arg keybase1.ResolveIdentifyImplicitTeamArg) (res keybase1.ResolveIdentifyImplicitTeamRes, err error) {
-	return res, fmt.Errorf("TODO: implement ResolveIdentifyImplicitTeam")
+	ctx = libkb.WithLogTag(ctx, "RII")
+	defer h.G().CTrace(ctx, "IdentifyHandler#ResolveIdentifyImplicitTeam", func() error { return err })()
+
+	h.G().Log.CDebugf(ctx, "ResolveIdentifyImplicitTeam assertions:'%v'", arg.Assertions)
+
+	if len(arg.Assertions) == 0 {
+		return res, fmt.Errorf("empty assertions")
+	}
+
+	split := strings.Split(arg.Assertions, "#")
+	if len(split) > 2 {
+		return res, fmt.Errorf("too many reader divison ('#') in assertions: %v", arg.Assertions)
+	}
+
+	writerAssertions, err := h.parseAssertionList(ctx, split[0])
+	if err != nil {
+		return res, err
+	}
+
+	var readerAssertions []libkb.AssertionExpression
+	if len(split) >= 2 && len(split[1]) > 0 {
+		readerAssertions, err = h.parseAssertionList(ctx, split[1])
+		if err != nil {
+			return res, err
+		}
+	}
+
+	return h.resolveIdentifyImplicitTeamHelper(ctx, arg, writerAssertions, readerAssertions)
+}
+
+func (h *IdentifyHandler) resolveIdentifyImplicitTeamHelper(ctx context.Context, arg keybase1.ResolveIdentifyImplicitTeamArg,
+	writerAssertions, readerAssertions []libkb.AssertionExpression) (res keybase1.ResolveIdentifyImplicitTeamRes, err error) {
+
+	lookupName := keybase1.ImplicitTeamDisplayName{
+		IsPublic: arg.IsPublic,
+	}
+	if len(arg.Suffix) > 0 {
+		lookupName.ConflictInfo, err = libkb.ParseImplicitTeamDisplayNameSuffix(arg.Suffix)
+		if err != nil {
+			return res, err
+		}
+	}
+
+	// TODO probably creates conflicts with duplicates such as "me,chris,chris" or "me,t_tracy,t_tracy@rooter".
+
+	var resolvedAssertions []resolvedAssertion
+
+	err = h.resolveImplicitTeamSet(ctx, writerAssertions, &lookupName.Writers, &resolvedAssertions)
+	if err != nil {
+		return res, err
+	}
+	err = h.resolveImplicitTeamSet(ctx, readerAssertions, &lookupName.Readers, &resolvedAssertions)
+	if err != nil {
+		return res, err
+	}
+
+	lookupNameStr, err := teams.FormatImplicitTeamDisplayName(context.TODO(), h.G(), lookupName)
```

Comment:
don't we have a context to pass here?

Reply:
Do we handle these cases?

* `bob,tom#tom` (should be the same as `bob,tom`)
* `bob,tom@github#tom` (should be the same as `bob,tom` if `tom@github` maps to `tom`)

Reply:
Nope, and handling that will be gross, I'd like to do it in a PR after this one.

Reply:
OK, I think it's worth doing...  Obviously you have to do that after the resolve step....

Reply:
Oh actually these are handled. Because inside `LookupOrCreateImplicitTeam` `libkb.ParseImplicitTeamDisplayName` does deduplication. So this layer would end up doing duplicate identifies, but would work. I added tests.